cmake/python.cmake

```diff
@@ -10,32 +10,26 @@ message(STATUS "Using PYTHON_EXECUTABLE: ${PYTHON_EXECUTABLE}")
 
 set(_PYTHON_PATH_VERSION_SUFFIX "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}")
 
-set(enable_setuptools_deb_layout OFF)
-if(EXISTS "/etc/debian_version")
-  set(enable_setuptools_deb_layout ON)
+# setuptools is fussy about windows paths, make sure the install prefix is in native format
+if(WIN32)
+  file(TO_NATIVE_PATH "${CMAKE_INSTALL_PREFIX}" SETUPTOOLS_INSTALL_PREFIX)
 endif()
-option(SETUPTOOLS_DEB_LAYOUT "Enable debian style python package layout" ${enable_setuptools_deb_layout})
 
-if(SETUPTOOLS_DEB_LAYOUT)
-  message(STATUS "Using Debian Python package layout")
-  set(PYTHON_PACKAGES_DIR dist-packages)
-  set(SETUPTOOLS_ARG_EXTRA "--install-layout=deb")
-  # use major version only when installing 3.x with debian layout
-  if("${PYTHON_VERSION_MAJOR}" STREQUAL "3")
-    set(_PYTHON_PATH_VERSION_SUFFIX "${PYTHON_VERSION_MAJOR}")
+# this block determines the same value as the Python function get_python_install_dir() from python/catkin/builder.py
+if(NOT DEFINED PYTHON_INSTALL_DIR)
+  execute_process(COMMAND "${PYTHON_EXECUTABLE}"
```

Comment:
I think that this pr covers the changes: https://github.com/ros/catkin/pull/832

I didn't find a need to refactor the `python.cmake` file after all. I can just call it with cmake in script mode and extract the information from the existing I/O. It is a bit fragile in that a change in the structure of that output could break the Python code, but it is covered in tests, so the tests would start failing in that case.